lib/matplotlib/tests/test_pyplot.py

```diff
@@ -484,3 +486,71 @@ def test_matshow():
 
     # Smoke test that matshow does not ask for a new figsize on the existing figure
     plt.matshow(arr, fignum=fig.number)
+
+
+def assert_signatures_identical(plt_meth, original_meth, remove_self_param=False):
```

Comment:
I don’t have time to check the details, but this functions seems quite heavy-weight now that we only check the really simple function set_loglevel. Do we really bread/want that complexity?

Reply:
It is a bit complex; we could extend the test to cover additional things in a later PR like `plt.rc`, `plt.getp`, `plt.figlegend`, `plt.imread`, `plt.imsave`, and maybe some others, but that does hit some limitations; I think the test as written doesn't handle overloads, and making it do that also might be more trouble than it's worth.

Reply:
Ah, I see. The complexity arises from ensuring identical typing.

As long as it's only about `set_loglevel()`, it would be sufficient for me to check identical names and numbers of arguments. That ensures identical runtime behavior. At the same time, this is probably the biggest risk: We modify the signature (renaming, adding arguments) and don't sync that. It's rather unlikely that we change the declared types - and even if that'd happen it's only a type-checking issue not a runtime issue.

Reply:
I have replaced with a low cost check. In addition to the names and numbers, I also check the kinds.